app/src/main/kotlin/compiler/intermediate/generators/FunctionDetailsGenerator.kt

```diff
@@ -10,12 +10,18 @@ interface FunctionDetailsGenerator : VariableAccessGenerator {
         val result: IFTNode?
     )
 
+    // both normal and generator functions, caller-side
     fun genCall(args: List<IFTNode>): FunctionCallIntermediateForm
 
+    // normal functions, callee-side
     fun genPrologue(): ControlFlowGraph
-
     fun genEpilogue(): ControlFlowGraph
 
+    // generators, callee-side
+    fun genInit(): ControlFlowGraph
+    fun genResume(): ControlFlowGraph
```

Comment:
And just to clarify: The prologue and epilogue will be used to build the resume function, which will contain the generator's actual code. So, basically, what I mean is that we should split `genResume` into `genResumePrologue` and `genResumeEpilogue`, but since these two will be very similar to `genPrologue` and `genEpilogue` of normal functions, we may as well reuse the latter and only add checks for `isGenerator` in them.